src/core/atom.ts

```diff
@@ -37,7 +37,7 @@ type Write<Update, Result extends void | Promise<void>> = (
   update: Update
 ) => Result
 
-interface WithInitialValue<Value> {
+export interface WithInitialValue<Value> {
```

Comment:
I think it should work with type alias as it used to be so:
```suggestion
type WithInitialValue<Value> = {
```
(but, eslint yells at this...)